src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -561,6 +561,16 @@ private void attemptConnectionUsingCachedTopology() throws SQLException {
     }
 
     this.hosts = cachedHosts;
+    try {
+      final String user = hostsList.get(0).getUser();
+      for(HostInfo hostInfo : this.hosts){
+        hostInfo.setUser(user);
+      }
+    }
+    catch(Exception e){
+      // Ignore, and use Cached information
+    }
+
```

Comment:
I don't think this chunk of code actually throws any exception? Unless it's a NullPointerException.
Why are you just getting the first user here?

Reply:
Made it more clear by grabbing the User from the connectionUrl instead.